### PR TITLE
raftstore: add commit log duration metrics

### LINKED ISSUE #5421: Add a metric to observe raft log commit duration
For now, we have no way to observe commit duration of a raft log, #3688 try to solve this by adding an extra `CLOCK_MONOTONIC_RAW`(syscall), but it turns out too expansive. We need to find a less expansive to observe the duration.

### PR DESCRIPTION
**Pull request by NingLin-P**
###  What have you changed?

add commit log duration metrics which spans from propose to committed

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

manual test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

pingcap/tidb-ansible#577

###  Refer to a related PR or issue link (optional)

close #5421

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -1318,6 +1318,11 @@ impl Peer {
                 if lease_to_be_updated {
                     let propose_time = self.find_propose_time(entry.get_index(), entry.get_term());
                     if let Some(propose_time) = propose_time {
+                        ctx.raft_metrics.commit_log.observe(duration_to_sec(
+                            (ctx.lease_time.unwrap_or_else(monotonic_raw_now) - propose_time)
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: why use lease time here?

Comment (NingLin-P) [reply src/raftstore/store/peer.rs]: you can see #5421 for the reason to use lease time

**Comment by siddontang** [comment]
```
            
test-4: < oltp_point_select >
    * QPS : 75282.48 ± 1.4787% (std=730.39) delta: 1.39%
    * AvgMs : 3.40 ± 1.4706% (std=0.03) delta: -1.39%
    * PercentileMs99 : 7.43 ± 0.0000% (std=0.00) delta: -1.72%
```

performance reduced

@NingLin-P

**Comment by siddontang** [comment]
em, why QPS increased?

**Comment by NingLin-P** [comment]
> em, why QPS increased?

don't know either ...

**Pull request merged**